tangchengjin/demo.py

```diff
@@ -2,37 +2,63 @@
 from random import choice
 name = input("Input file name: ")  ## optional: loop
 ## 1.1. try.. except check file exists
+flag = 0
+while flag == 0 :
+    try :
+        fhand = open(name)
+        print(fhand)
+        flag = 1
+    except :
+        name = input("Can't find the file, please enter the file name again: ")
 
-## N = input("How may grams: ")
-## N = 3
+
+N = int(input("How may grams: "))
 ## 2. N-gram instead of 3
 fileList = open(name).read().split()
-## newline ?
 d = {}
 for i in range(len(fileList)):
-    ## 2.1 windows length = N - 1
-    word1 = fileList[i]
-    word2 = fileList[(i + 1) % len(fileList)]
-    key = (word1, word2)
-    value = fileList[(i + 2) % len(fileList)]
-    if key in d:
-        d[key].append(value)
+    #word1 = fileList[i]
+    #word2 = fileList[(i + 1) % len(fileList)]
+    #key = (word1, word2)
+    key = []
+    for j in range(N - 1):
+        key.append(fileList[(i + j) % len(fileList)])
+    value = fileList[(i + j + 1) % len(fileList)]
+    keyTuple = tuple(key)
+    if keyTuple in d:
+        d[keyTuple].append(value)
     else:
-        d[key] = [value]
+        d[keyTuple] = [value]
 
-num_words_to_gen = int(input("How many words to generate: "))
+num_words_to_gen = input("How many words to generate: ")
 ## 1.2 "q" to quit, loop
 ##  check int convert
+nWords = 0
+while nWords == 0:
```

Comment:
Here, the loop is supposed to ask for how many words to generate over and over again, i.e., 
`How many words to generate: 7
... he who commands the sea commands everything ...
How many words to generate: 9
... to be or not to be, this is the ...
How many words to generate: q
Finished.`
Currently, your code will result in an error if the user inputs 'q', because `break` only breaks out of the loop, not the whole program. Current code doesn't repeatedly generate random texts either.